infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -312,6 +319,12 @@ private ServerContext getServerContext(Map<String, Object> opts) {
      * @param serverContext server context (server information)
      */
     private void startDiscoveryService(ServerContext serverContext) {
+        // instantiate clusterManager to find if the deployment is a SaaS deployment. This is used for computing the
+        // localCorfuEndpoint for the corfu runtime.
+        // This instance of clusterManager is further passed to discoveryService
+        CorfuReplicationClusterManagerAdapter clusterManagerAdapter =
+                getClusterManagerAdapter(serverContext.getPluginConfigFilePath());
+        CorfuSaasEndpointProvider.init(serverContext.getPluginConfigFilePath(), clusterManagerAdapter.isSaasDeployment());
```

Comment:
actually.. @hisundar and I were discussing about this a few weeks ago.  isSassDeployment will be available from the topology.  What if a topology is not available when this call is made?

Sundar's suggestion was also that relying on the cluster manager plugin to know if a deployment is saas may not be optimal and he had made a change on the CL AFAIK.

Reply:
discussed offline. isSassDeployment would be set during compile time by the client, and LR access it by using the client's CorfuReplicationClusterManagerAdapter